app/Http/Controllers/CalendarController.php

```diff
@@ -205,4 +209,16 @@ public function today(Calendar $calendar)
         return redirect()->back()
             ->with('success', __('calendars.edit.today'));
     }
+
+    /**
+     * @param Campaign $campaign
+     * @param Calendar $calendar
+     * @return \Illuminate\Http\Resources\Json\AnonymousResourceCollection
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function eventLength(Calendar $calendar, ValidateLength $request)
+    {
+        $this->authorize('view', $calendar);
+        return $this->lengthValidatorService->validateLength($calendar, $request);
```

Comment:
You can use `return response()->json($this->lengthValidator...)` for a cleaner separation of concern